kubeportal/api/views.py

```diff
@@ -36,31 +37,26 @@ def get_queryset(self):
             else:
                 raise Http404
 
-    # def partial_update(self, request, *args, **kwargs):
-    #     pk = int(self.kwargs['pk'])
-
-    #     if pk != self.request.user.pk:
-    #         if User.objects.filter(pk=pk).exists():
-    #             raise PermissionDenied
-    #         else:
-    #             raise Http404
-
-    #     target_user = User.objects.get(pk=pk)
-
-    #     if len(request.data) == 0:
-    #         logger.warning(f"Got empty body in patch request for user {target_user}.")
-    #         return Response(status=status.HTTP_422_UNPROCESSABLE_ENTITY)
-    #     else:
-    #         serializer = UserSerializer(target_user, data=request.data, partial=True) 
-    #         if serializer.is_valid():
-    #             serializer.save()
-    #             response = JsonResponse(data=serializer.data)
-    #             response.status_code = 204
-    #             return response
-    #         else:
-    #             logger.warning(f"Got invalid body in patch request for user {target_user}.")
-    #             return Response(status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+    def partial_update(self, request, *args, **kwargs):
+        pk = int(self.kwargs['pk'])
+        if pk != self.request.user.pk:
+            if User.objects.filter(pk=pk).exists():
+                raise PermissionDenied
+            else:
+                raise Http404
 
+        target_user = User.objects.get(pk=pk)
+        if len(request.data) == 0:
+            logger.warning(f"Got empty body in patch request for user {target_user}.")
+            return Response(status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+        else:
+            serializer = UserSerializer(target_user, data=request.data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+                return JsonResponse(data=serializer.data, status=200)
+            else:
+                logger.warning(f"Got invalid body in patch request for user {target_user}.")
+                return Response(status=status.HTTP_422_UNPROCESSABLE_ENTITY)
```

Comment:
Just commented in? Any change here?

Reply:
I use status=200 within the JsonResponse constructor.  and I use 200 instead of 204 since we return the updated user object.... and I don't know, I've experienced scary things in the meanwhile and don't know why they are solved now. "Socket Connection reset by peer", for example. In the end, your solution was kind of right..